spdx-utils/build.gradle.kts

```diff
@@ -0,0 +1,415 @@
+                            if (isGpl) {
+                                // Note: Do not use appendln() here as that would write out platform-native line endings, but we
+                                // want to normalize on Unix-style line endings for consistency.
+                                append("\n")
+
+                                // For GPL the "or later version" text is just an amendment that reads better as a prefix as then no
+                                // text follows the license's final "That's all there is to it!" sentence.
+                                append(SpdxLicense::class.java.getResource("/$resourcePath/${'$'}baseId").readText())
+                            }
+                        } else {
+                            append(SpdxLicense::class.java.getResource("/$resourcePath/${'$'}id").readText())
+                        }
+                    }
+                }
+            }
+
+        """.trimIndent())
+    } else {
+        enumFile.appendText("""
+                /**
+                 * The full $description text as a string.
+                 */
+                val text by lazy { javaClass.getResource("/$resourcePath/${'$'}id").readText() }
+            }
+
+        """.trimIndent())
+    }
+
+    logger.quiet("Generated SPDX $description enum file '$enumFile'.")
+
+    return ids
+}
+
+fun generateLicenseTextResources(description: String, ids: Map<String, LicenseMetaData>, resourcePath: String) {
+    logger.quiet("Determining SPDX $description texts...")
+
+    val scanCodeLicensePath = "$buildDir/SvnExport/licenses/scancode-toolkit"
+    val spdxIdToScanCodeKey = mutableMapOf()
+
+    file(scanCodeLicensePath).listFiles(FileFilter { it.name.endsWith(".yml") }).forEach { file ->
+        file.readLines().forEach { line ->
+            val keyAndValue = line.split(Regex("^spdx_license_key:"), 2)
+            if (keyAndValue.size == 2) {
+                spdxIdToScanCodeKey[keyAndValue.last().trim()] = file.name.removeSuffix(".yml")
+            }
+        }
+    }
+
+    val resourcesDir = file("src/main/resources/$resourcePath").apply {
+        if (isDirectory && !deleteRecursively()) {
+            throw GradleException("Failed to delete the existing '$this' directory.")
+        }
+
+        mkdirs()
+    }
+
+    ids.forEach { id, meta ->
+        val resourceFile = resourcesDir.resolve(id)
+
+        // Prefer the texts from ScanCode as these have better formatting than those from SPDX.
+        val candidates = mutableListOf(
+                "$scanCodeLicensePath/${spdxIdToScanCodeKey[id]}.LICENSE",
+                "$buildDir/SvnExport/licenses/spdx/$id.txt"
+        )
+
+        if (meta.deprecated) {
+            candidates += "$buildDir/SvnExport/licenses/spdx/deprecated_$id.txt"
+        }
+
+        val i = candidates.iterator()
+        while (true) {
+            if (i.hasNext()) {
+                val candidate = i.next()
```

Comment:
This while(true) construct can be re-written a bit simpler as:

for (candidate in candidates.iterator()) {
    try {
         ...
         return
    } catch () { ... }
}
throw ...

Reply:
Alright, makes sense.